### ISSUE TITLE
Confusing name: allowFlowReversal

### ISSUE THREAD

**Issue by Mathadon** [iea-annex60/modelica-annex60#531]
I have encountered several occasions where people erroneously assume that `allowFlowReversal=false` implies that the flow _cannot_ reverse, while it _should_ not reverse. I.e. the user should still make sure himself that the flow does not reverse.
If the flow does reverse the equations lead to an exponential behaviour, causing the temperatures to explode.

We can counter this by implementing one or more of the following measures 1) renaming `allowFlowReversal`, due to backwards compatibility issues this does not seem feasible, 2) updating the comment of `allowFlowReversal` 3) update documentation, 4) add asserts that ensure that the flow does not reverse.

**Comment by mwetter** [comment]
I suggest options 2 and 3. We could do an assert in some components, but I would not do it in the base classes unless we can show that there is negligible overhead for translation and simulation time.

**Comment by Mathadon** [comment]
For 2) I propose to change

```"= true to allow flow reversal, false restricts to design direction (port_a -> port_b)"```

to

```"= false to simplify equations assuming (i.e. does not enforce!) that no flow reversal occurs"```

I dropped the hint that the flow should go from port_a to port_b. This should however be clear from the graphical illustration.

For 3) we could make the check conditional, with default disabled?

**Comment by mwetter** [comment]
For 2), I suggest a slight rewording (and removing the exclamation mark):
```
"= false to simplify equations, assuming, but not enforcing, no flow reversal"
```
For 3), could you briefly check overhead if enabled on reasonably large models. If there is negligible overhead, we could have it enabled by default, as I don't expect that the users who use it the wrong way, maybe because of not reading the documentation, will take the effort to enable it.

**Comment by Mathadon** [comment]
Note that we already have

```
  assert(m_flow > -m_flow_small or allowFlowReversal,
      "Reverting flow occurs even though allowFlowReversal is false");
```

in ```Annex60.Fluid.Interfaces.PartialTwoPortTransport```

**Comment by Mathadon** [comment]
I ran my model with and without adding a check in `PartialTwoPortInterface`. Even though it is hard to see the difference due to random computer load, there was a difference in simulation time of about 5%. 

I did an analysis off the asserts in my model, I have:
 - 250 asserts checking temperature bounds 
 - 230 asserts checking absolute humidity
 - 38 asserts checking absolute pressure
 - 581 asserts checking mass flow rate
 - 80 other

Of those 581 asserts 405 were introduced by the change I made.

While going through the asserts it also seems that 'alias asserts' are not removed.
